### ISSUE TITLE
Debugger/Observatory evaluateInFrame doesn't work inside methods using await

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#27167]
When at a breakpoint in an async method, we're unable to get values from `evaluateInFrame`. This makes debugging async code somewhat difficult.

@devoncarew thinks this might be a known issue, but I couldn't find an existing case about it.

I appreciate this case is quite light, so if this isn't totally obvious to you and you want a concrete example,. please let me know.

**Comment by devoncarew** [comment]
For a bit more context, this hits us in IntelliJ as well - we use expression evaluation to implement debugger tooltips and watchpoints.

**Comment by rmacnak-google** [comment]
Can you be more specific about how it fails? Does the attached VM crash, does the evaluation fail to happen, or does the evaluation say some value you think should be in scope is not in scope?

**Comment by DanTup** [comment]
Sorry, we thought this might be a known issue so didn't dig too much. If @devoncarew doesn't beat me to it, I'll do some more digging over the weekend and try to provide some more useful info.

**Comment by devoncarew** [comment]
Yup, here's a case that works:

```dart
main() async {
  var foo = 'bar';
  print(foo); // breakpoint here
}
```

If you break at the above line, you can get results from expression evaluation; `foo` returns `'bar'`, and `1+2` return `3`.

For this example:

```dart
import 'dart:async';

main() async {
  await new Future.value();
  var foo = 'bar';
  print(foo); // breakpoint here
}
```

expression evaluation does not work. The results you get (from `foo` or `1+2`) are:

```
==> {"id":"14","method":"evaluateInFrame","params":{"isolateId":"isolates/201098417","frameIndex":0,"expression":"foo"}}
<== {"jsonrpc":"2.0", "result":{"type":"@Error","_vmType":"LanguageError","kind":"LanguageError","id":"objects\/26","message":"'evaluate:source': error: line 1 pos 2: parameter name or type expected\n(:async_op, foo) => foo;\n ^\n"},"id":"14"}
```

It seems like it's not the fact that the method is `async` which causes the issue, but that you use `await` in the method.

**Comment by DanTup** [comment]
Possibly related - in my testing it doesn't seem to pause on exceptions in these methods unless I pause on all exceptions instead of just uncaught.

**Comment by DanTup** [comment]
@rmacnak-google Awesome, thanks! I'm not familiar with the release process and there's no version milestone/tag; if there's a 1.19.x would this be included or will it only be in 1.20?
